### PR TITLE
net: openthread: upmerge 20210205

### LINKED ISSUE #31938: Invalid SPDX license identifier used in file
**Describe the bug**
An invalid SPDX license identifier is being specified in
https://github.com/zephyrproject-rtos/openthread/blob/zephyr/third_party/NordicSemiconductor/libraries/nrf_security/config/nrf-config.h

`SPDX-License-Identifier: BSD-3-Clause OR Arm’s non-OSI source license`

The phrase "Arm's non-OSI source license" is not a valid SPDX license identifier.

**To Reproduce**
The problem is in the source files, and needs to be addressed there.

**Expected behavior**
Valid SPDX license identifier from https://spdx.org/licenses/ should be used, or 
a "LicenseRef-" identifier should be created and used.  Details on the "LicenseRef-" syntax can be found in https://spdx.github.io/spdx-spec/6-other-licensing-information-detected/#61-license-identifier should be used in the expression

**Impact**
Serious - this breaks tooling working to summarize the licenses for the sources and generate SPDX documents.

**Logs and console output**
```
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/config/nrf-config.h
SPDXID: SPDXRef-File-nrf-config.h
FileChecksum: SHA1: e667be982ab96d6d59de7b965538cb109d91d18c
FileChecksum: SHA256: fb5342294f7c89fba780af925abb41aecf583a1d692e73d05a8d0b2af581203c
LicenseConcluded: BSD-3-Clause OR Arm’s non-OSI source license
LicenseInfoInFile: Arm’s
LicenseInfoInFile: BSD-3-Clause
LicenseInfoInFile: license
LicenseInfoInFile: non-OSI
LicenseInfoInFile: source
FileCopyrightText: NOASSERTION

```
**Environment (please complete the following information):**
The problem can be seen in the source file. 

**Additional context**
This same problem can also be seen in:
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/ccm_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/chacha20_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/chachapoly_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/ecp_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/platform_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/poly1305_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/sha1_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/sha256_alt.h
FileName: ./modules/lib/openthread/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/threading_alt.h

### PR DESCRIPTION
**Pull request by edmont**
OpenThread upmerge up to commit `add99687` .

Fixes #31938.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

**Comment by carlescufi** [comment]
This would fix #31938, but before we decide to merge this I'd like @rlubos to see if we could instead just cherry-pick the license commit.

**Comment by rlubos** [comment]
@carlescufi Of course it's possible, personally though I'd prefer an upmerge - it's pretty small this time (we had an upmerge not long ago) and it'll keep the history clean.

**Comment by carlescufi** [comment]
> @carlescufi Of course it's possible, personally though I'd prefer an upmerge - it's pretty small this time (we had an upmerge not long ago) and it'll keep the history clean.

In that case I suggest we complete this.

**Review by carlescufi — APPROVED** [review]

File: west.yml
Comment (rlubos) [inline west.yml:115]: Please update SHA: `1d668284a0b897bd6d539e7c3ed8009f0028099d`

**Comment by nashif** [comment]
> @carlescufi Of course it's possible, personally though I'd prefer an upmerge - it's pretty small this time (we had an upmerge not long ago) and it'll keep the history clean.


```
# git diff  aad6518..1d66828 --stat
..
..
 168 files changed, 7752 insertions(+), 958 deletions(-)
```

how is this small?

**Comment by edmont** [comment]
> how is this small?

We are used to wider upmerges:

https://github.com/zephyrproject-rtos/openthread/commit/aad6518ee30d3840411b21437d396f0405e947c3
https://github.com/zephyrproject-rtos/openthread/commit/c3362bc8e24f7e1ec21242c03611595e2d8ec2f0

**Review by rlubos — APPROVED** [review]

**Comment by nashif** [comment]
> We are used to wider upmerges:

right, but this is supposed to fix an issue with licensing and we are in code freeze phase already.

**Pull request merged**
